OpenRA.Mods.RA/Activities/Infiltrate.cs

```diff
@@ -48,10 +48,14 @@ protected override void OnInside(Actor self)
 			foreach (var t in target.TraitsImplementing<INotifyInfiltrated>())
 				t.Infiltrated(target, self);
 
-			self.Dispose();
+			if (enterBehaviour == EnterBehaviour.Suicide)
+				self.Kill(self);
+			else if (enterBehaviour == EnterBehaviour.Dispose)
+				self.Dispose();
 
 			if (target.Info.HasTraitInfo<BuildingInfo>())
-				Game.Sound.PlayToPlayer(self.Owner, "bldginf1.aud");
+				Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech",
```

Comment:
This is no longer tied to the building infiltration file, so perhaps this check should be moved to the infiltration trait, and if a notification shouldn't be played, pass null instead of the file name to the activity's constructor.